src/mlpack/core/util/param.hpp

```diff
@@ -56,23 +53,117 @@ using DatasetInfo = DatasetMapper<IncrementPolicy, std::string>;
- *     documentation.  See the SEE_ALSO() macro.  This is a varargs argument, so
- *     you can add as many SEE_ALSO()s as you like.
  */
-#define PROGRAM_INFO(NAME, SHORT_DESC, DESC, ...) \
-    static mlpack::util::ProgramDoc \
-    io_programdoc_dummy_object = mlpack::util::ProgramDoc(NAME, SHORT_DESC, \
-    []() { return DESC; }, { __VA_ARGS__ } )
+#define BINDING_SHORT_DESC(SHORT_DESC) static \
+    mlpack::util::ShortDescription \
+    io_programshort_desc_dummy_object = mlpack::util::ShortDescription( \
+    SHORT_DESC);
+
+/**
+ * Document an executable.  Only one instance of this macro should be
+ * present in your program!  Therefore, use it in the main.cpp
+ * (or corresponding executable) in your program.
+ *
+ * @see mlpack::IO, PARAM_FLAG(), PARAM_INT_IN(), PARAM_DOUBLE_IN(),
+ * PARAM_STRING_IN(), PARAM_VECTOR_IN(), PARAM_INT_OUT(), PARAM_DOUBLE_OUT(),
+ * PARAM_VECTOR_OUT(), PARAM_INT_IN_REQ(), PARAM_DOUBLE_IN_REQ(),
+ * PARAM_STRING_IN_REQ(), PARAM_VECTOR_IN_REQ(), PARAM_INT_OUT_REQ(),
+ * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
+ *
+ * @param LONG_DESC Long string describing what the program does. Newlines
+ *     should not be used here; this is taken care of by IO (however, you
+ *     can explicitly specify newlines to denote new paragraphs).  You can
+ *     also use printing macros like PRINT_PARAM_STRING(), PRINT_DATASET(),
+ *     and others.
+ */
+#define BINDING_LONG_DESC(LONG_DESC) static \
+    mlpack::util::LongDescription \
+    io_programlong_desc_dummy_object = mlpack::util::LongDescription( \
+    []() { return std::string(LONG_DESC); });
 
+/**
+ * Document an executable.  Mutiple instance of this macro can be
+ * present in your program!  Therefore, use it in the main.cpp
+ * (or corresponding executable) in your program.
+ *
+ * @see mlpack::IO, PARAM_FLAG(), PARAM_INT_IN(), PARAM_DOUBLE_IN(),
+ * PARAM_STRING_IN(), PARAM_VECTOR_IN(), PARAM_INT_OUT(), PARAM_DOUBLE_OUT(),
+ * PARAM_VECTOR_OUT(), PARAM_INT_IN_REQ(), PARAM_DOUBLE_IN_REQ(),
+ * PARAM_STRING_IN_REQ(), PARAM_VECTOR_IN_REQ(), PARAM_INT_OUT_REQ(),
+ * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
+ *
+ * @param EXAMPLE Long string describing a simple usage example.. Newlines
+ *     should not be used here; this is taken care of by IO (however, you
+ *     can explicitly specify newlines to denote new paragraphs).  You can
+ *     also use printing macros like PRINT_CALL(), PRINT_DATASET(),
+ *     and others.
+ */
+#ifdef __COUNTER__
+  #define BINDING_EXAMPLE(EXAMPLE) static \
+      mlpack::util::Example \
+      JOIN(io_programexample_dummy_object_, __COUNTER__) = \
+      mlpack::util::Example( \
+      []() { return(std::string(EXAMPLE)); });
+#else
+  #define BINDING_EXAMPLE(EXAMPLE) static \
+      mlpack::util::Example \
+      JOIN(JOIN(io_programexample_dummy_object_, __LINE__), opt) = \
+      mlpack::util::Example( \
+      []() { return(std::string(EXAMPLE)); });
+#endif
+/**
```

Comment:
```suggestion
#endif

/**
```
Just a simple spacing fix. :)